scripts/check_deps.sh

```diff
@@ -35,20 +35,33 @@ missing_dep() {
 }
 
 GO_DEPS=(
-    "$GO_BIN/stringer"
-    "$GO_BIN/msgp"
-    "$GO_BIN/golangci-lint"
+    "msgp"
+    "golangci-lint"
+    "oapi-codegen"
+    "swagger"
 )
 
+check_go_binary_version() {
+  binary_name=$1
+  expected_version=$(grep "$binary_name" scripts/buildtools/versions | awk '{print $2}')
+  actual_version=$(go version -m "$GO_BIN/$binary_name" | awk 'NR==3 {print $3}')
+
+  if [ "$expected_version" != "$actual_version" ]; then
+      echo "$YELLOW_FG[WARNING]$END_FG_COLOR $binary_name version mismatch, expected $expected_version, but got $actual_version"
+  fi
+}
+
 check_deps() {
-    for path in ${GO_DEPS[*]}
+    for dep in ${GO_DEPS[*]}
     do
-        if [ ! -f "$path" ]
+        if [ ! -f "$GO_BIN/$dep" ]
         then
             # Parameter expansion is faster than invoking another process.
             # https://www.linuxjournal.com/content/bash-parameter-expansion
-            missing_dep "${path##*/}"
+            missing_dep "${dep##*/}"
         fi
+
+        check_go_binary_version "$dep"
```

Comment:
Maybe this is ok as is, but the script won't exit with an error code, so the build will continue, it will just have this message at the tope.